source/MaterialXGenShader/Shader.cpp

```diff
@@ -52,42 +52,7 @@ void Shader::initialize(ElementPtr element, ShaderGenerator& shadergen, const Ge
             if (_rootGraph->isEditable(*inputSocket))
             {
                 // Create the uniform
-                createUniform(PIXEL_STAGE, PUBLIC_UNIFORMS, inputSocket->type, inputSocket->name, EMPTY_STRING, inputSocket->value);
-            }
-        }
-    }
-    
-    // Check if a complete interface is requested
-    if (options.shaderInterfaceType == SHADER_INTERFACE_COMPLETE)
-    {
-        // Create uniforms for all node inputs that has not been connected already
-        for (ShaderNode* node : _rootGraph->getNodes())
-        {
-            for (ShaderInput* input : node->getInputs())
-            {
-                if (!input->connection)
-                {
-                    // Check if the type is editable otherwise we can't 
-                    // publish the input as an editable uniform.
-                    if (input->type->isEditable() && node->isEditable(*input))
-                    {
-                        // Use a consistent naming convention: <nodename>_<inputname>
-                        // so application side can figure out what uniforms to set
-                        // when node inputs change on application side.
-                        const string interfaceName = node->getName() + "_" + input->name;
-
-                        ShaderGraphInputSocket* inputSocket = _rootGraph->getInputSocket(interfaceName);
-                        if (!inputSocket)
-                        {
-                            inputSocket = _rootGraph->addInputSocket(interfaceName, input->type);
-                            inputSocket->value = input->value;
-                        }
-                        inputSocket->makeConnection(input);
-
-                        // Create the uniform
-                        createUniform(PIXEL_STAGE, PUBLIC_UNIFORMS, inputSocket->type, inputSocket->name, EMPTY_STRING, inputSocket->value);
-                    }
-                }
+                createUniform(PIXEL_STAGE, PUBLIC_UNIFORMS, inputSocket->type, inputSocket->variable, EMPTY_STRING, inputSocket->value);
```

Comment:
Since the graph input sockets have already been created (either for "reduced" or "complete" mode), we can just create a uniform for each one here.